datasets/pg19/pg19.py

```diff
@@ -0,0 +1,155 @@
+"""TODO(pg19): Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import json
+import os
+
+import tensorflow as tf
+
+import nlp
+
+
+# TODO(pg19): BibTeX citation
+_CITATION = """\
+@article{raecompressive2019,
+  author = {Rae, Jack W and Potapenko, Anna and Jayakumar, Siddhant M and
+            Hillier, Chloe and Lillicrap, Timothy P},
+  title = {Compressive Transformers for Long-Range Sequence Modelling},
+  journal = {arXiv preprint},
+  url = {https://arxiv.org/abs/1911.05507},
+  year = {2019},
+}
+
+"""
+
+# TODO(pg19):
+_DESCRIPTION = """\
+This repository contains the PG-19 language modeling benchmark.
+It includes a set of books extracted from the Project Gutenberg books library, that were published before 1919.
+It also contains metadata of book titles and publication dates.
+"""
+
+_ROOT_URL = "gs://deepmind-gutenberg/"
+_METADATA_URL = os.path.join(_ROOT_URL, "metadata.csv")
+
+
+class Pg19(nlp.GeneratorBasedBuilder):
+    """TODO(pg19): Short description of my dataset."""
+
+    # TODO(pg19): Set up version.
+    VERSION = nlp.Version("0.1.0")
+
+    def _info(self):
+        # TODO(pg19): Specifies the nlp.DatasetInfo object
+        return nlp.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # nlp.features.FeatureConnectors
+            features=nlp.Features(
+                {
+                    "short_book_title": nlp.Value("string"),
+                    "publication_date": nlp.Value("int32"),
+                    "url": nlp.Value("string"),
+                    "text": nlp.Value("string"),
+                    # These are the features of your dataset like images, labels ...
+                }
+            ),
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage="https://github.com/deepmind/pg19",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO(pg19): Downloads the data and defines the splits
+        # dl_manager is a nlp.download.DownloadManager that can be used to
+        # download and extract URLs
+
+        def get_filename(path):
+            return os.path.splitext(os.path.basename(path))[0]
+
+        def download_listdir(url, local_filepath):
```

Comment:
same as here?

Reply:
I actually think it's good with leaving the function in here since it's only going to be used inside the function.